android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/scenario/LocalScenario.java

```diff
@@ -0,0 +1,403 @@
+                mIoHandler.post(mAuthRepository::stop);
+            }
+        }
+
+        @Override
+        public void onSessionError() {
+            Log.w(TAG, "MobileWalletAdapter session error");
+            stopNoConnectionTimer();
+            if (mClientCount.decrementAndGet() == 0) {
+                synchronized (mLock) {
+                    mActiveAuthorization = null;
+                }
+                mIoHandler.post(mCallbacks::onScenarioServingComplete);
+                mIoHandler.post(mAuthRepository::stop);
+            }
+        }
+    };
+
+    private final MobileWalletAdapterServer.MethodHandlers mMethodHandlers =
+            new MobileWalletAdapterServer.MethodHandlers() {
+        @Override
+        public void authorize(@NonNull MobileWalletAdapterServer.AuthorizeRequest request) {
+            // Clear the active auth token immediately upon beginning authorization; it will be
+            // set to valid only on successful completion
+            synchronized (mLock) {
+                mActiveAuthorization = null;
+            }
+
+            final NotifyingCompletableFuture<AuthorizeRequest.Result> future = new NotifyingCompletableFuture<>();
+            future.notifyOnComplete(f -> mIoHandler.post(() -> { // Note: run in IO thread context
+                try {
+                    final AuthorizeRequest.Result authorize = f.get(); // won't block
+
+                    if (authorize != null) {
+                        final String name = request.identityName != null ? request.identityName : "";
+                        final Uri uri = request.identityUri != null ? request.identityUri : Uri.EMPTY;
+                        final Uri relativeIconUri = request.iconUri != null ? request.iconUri : Uri.EMPTY;
+                        final AuthRecord authRecord = mAuthRepository.issue(
+                                name, uri, relativeIconUri, authorize.publicKey,
+                                authorize.accountLabel, request.cluster, authorize.walletUriBase,
+                                authorize.scope);
+                        Log.d(TAG, "Authorize request completed successfully; issued auth: " + authRecord);
+                        synchronized (mLock) {
+                            mActiveAuthorization = authRecord;
+                        }
+
+                        final String authToken = mAuthRepository.toAuthToken(authRecord);
+                        request.complete(new MobileWalletAdapterServer.AuthorizationResult(
+                                authToken, authorize.publicKey, authorize.accountLabel,
+                                authorize.walletUriBase));
+                    } else {
+                        request.completeExceptionally(new MobileWalletAdapterServer.RequestDeclinedException(
+                                "authorize request declined"));
+                    }
+
+                    startNoConnectionTimer();
```

Comment:
Curious: what necessitates starting the no connection timer here (and elsewhere below, in response to `MobileWalletAdapterServer.MethodHandlers` callbacks)?

Reply:
did you see [this comment](https://github.com/solana-mobile/mobile-wallet-adapter/pull/363#issuecomment-1430769129)?

I can reproduce this case where the initial connection succeeds and we get a valid request (even tho we are on low power mode) but subsequent operations (like, user confirming authorization) will hang. 

I have logged the web socket server to see if we receive _anything_ after the initial request is received that would tell us that the connection is closed/paused and I am not seeing anything.

Reply:
I can make a video that shows this if youd like, but steps to repo are the same:
- have device in low power
- open dapp and initiate a request 
- if you:
  - **do** see the request land, try to complete it (tap authorize), the request completion will hang
  - **do not** see the connection request, you should see the low power warning. try these steps again to see the other case

On saga I am seeing the first case more often than the second so I think its important to fix.

Reply:
What I'm worried about is that, once the session is open, there is no requirement on how long between actions the dApp is allowed to take. Let me think about this a little...